### DISCUSSION TITLE
Fail to build the page on my github

### DISCUSSION THREAD

**Discussion by XIYANG1018** [greenelab/lab-website-template#233] [Q&A] [answered]
Hi all!

I didn't change anything in the code, jusr simply forked the template and then changed the settings to main, root, but it always fails to build the page. Can someone help me with this?? Thanks!
![image](https://github.com/greenelab/lab-website-template/assets/117253597/5a24256d-ee79-4c4a-b463-2ac57c1394b5)
![image](https://github.com/greenelab/lab-website-template/assets/117253597/9e553a7b-a5a9-4c9b-b227-3fceb375eea3)
![image](https://github.com/greenelab/lab-website-template/assets/117253597/3b5766d8-b197-47d9-b219-89df9317ec12)
![image](https://github.com/greenelab/lab-website-template/assets/117253597/bc2f9b0a-3421-4d4f-b6fb-e294327c9dce)
![image](https://github.com/greenelab/lab-website-template/assets/117253597/8d814b8d-157b-40c2-8fba-0bcbb648f738)

**Comment by vincerubinetti** [comment]
Here is the repo in question. In the future please provide a link to you repo, as I need that to help you in almost all cases.
https://github.com/XIYANG1018/lab-website-template/actions/runs/7577131747/job/20637347112

```
Warning:  github-pages can't satisfy your Gemfile's dependencies.

  Liquid Exception: Liquid syntax error (/github/workspace/_includes/head.html line 22): Expected end_of_string but found id included in /_layouts/default.html
```

Did you [follow all of the setup instructions here](https://greene-lab.gitbook.io/lab-website-template-docs/getting-started/set-up-your-site)?

**Reply by vincerubinetti** [reply, marked as answer]
Looking at your gh-acitons workflow run history, let's take it step by step what happened.

First you got an error `❌ pages build and deployment with artifacts-next`. Within that, the step that was failing was `❌ Build with Jekyll`. I know this is confusing, but this is _GitHub_ trying to build the site with Jekyll with its default setup, not the _template_ trying to build the site with Jekyll. For a long time, GitHub Pages has had "built-in" support for _basic_ Jekyll sites. With version 1.0 of the template, we started using Jekyll features beyond the basic ones that GitHub allowed by default, so now the template has to build your site itself (`build-site` workflow). However, GitHub still assumes your repo is a _basic_ Jekyll site and tries to build it as such (in `pages-build-deployment` workflow), _unless you tell it not to_ by putting a `.nojekyll` file on the `gh-pages` branch. The template's `first-time-setup` workflow does this for you, but you hadn't run it.

After that, I see you did run `first-time-setup`, which successfully put the `.nojekyll` file in your `gh-pages` branch. But before that you made several changes to the `Gemfile` like `gem 'faraday-retry', '~> 2.0'`, presumably following the error message in the logs. This caused the next run that would've built your site (in ["on pages deploy" → `build-site`](https://github.com/XIYANG1018/lab-website-template/actions/runs/7577528260)) to fail, due to package version incompatibilities.

You should revert the changes made in [this commit](https://github.com/XIYANG1018/lab-website-template/commit/013eb5fea96b212d31b9ef64df713e06c535cc7f) and [this commit](https://github.com/XIYANG1018/lab-website-template/commit/c18a1fb4104b5fd97c41dda2d09a7e10b9f6c44a), then hopefully your site should build. Also, [this commit](https://github.com/XIYANG1018/lab-website-template/commit/5b7d446fdc6313a3671678852b114e8fdcba9dec) has `testbed.html` in it which you do not need.

Btw just to make sure something didn't change with GitHub Actions or Ruby Gem versions, I forked the template fresh to my personal account, followed the setup instructions, and it works:
https://github.com/vincerubinetti/lab-website-template/actions
https://vincerubinetti.github.io/lab-website-template/

**Reply by XIYANG1018** [reply]
Thank you very much Vincent! Yes! I deleted the original repo of my site and forked from your template following the setup instructions and it did work! I am still trying to fix the problem you mentioned in the message to see if the original site can also be built. Your reply helps a lot!

**Reply by vincerubinetti** [reply]
Starting from scratch is probably the best idea. Then you can just copy over the config and text and pictures you already had in the first repo.
